### DISCUSSION TITLE
Is it possible to have multiple fields to be treated as unique when they occur together?

### DISCUSSION THREAD

**Discussion by RitikDua** [prisma/prisma#3312] [Q&A] [answered]
Let just say i have a datamodel of 'cart':
```
model Cart{
  id Int @id @default(autoincrement())
  product Product @relation(name:"CartProduct",references: [id],fields:[productId])
  quantity Int
  productId Int  
  user    User   @relation(fields: [userId], references: [id])
  userId  Int    
}
```
I would like to treat the cart as unique only when **userId and productId occurs together.**

**Comment by janpio** [comment, marked as answer]
Yes of course: https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/data-model#unique, https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/data-model#multi-field-unique-attributes to be more exact is what you are looking for.

```
model User {
  id        Int     @default(autoincrement())
  firstName String
  lastName  String
  isAdmin   Boolean @default(false)
  @@unique([firstName, lastName])
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
